fullcontact/api/base/enrich_base.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+"""
+This module serves the base class for all
+FullContact Enrich API request classes.
+"""
+
+from abc import ABCMeta, abstractmethod
+from concurrent.futures import Future
+
+from .base import ApiBase
+
+
+class EnrichBase(ApiBase, metaclass=ABCMeta):
+
+    @property
+    @abstractmethod
+    def _enrich_endpoint(self) -> str:
+        r"""
+        Endpoint for enrich.
+
+        :return: Enrich endpoint that can be used by self._enrich_url
+        """
+        raise NotImplementedError(
+            "The property '_enrich_endpoint' is "
+            "not implemented in %s" % self.__class__
+        )
+
+    @property
+    @abstractmethod
+    def _enrich_request_handler(self):
+        r"""
+        Handler for the Enrich API request.
+        This has to be the instance of a class which provides a validate()
+        method to validate the enrich request
+        """
+        raise NotImplementedError(
+            "The property '_enrich_request_handler' is "
+            "not implemented in %s" % self.__class__
+        )
+
+    @property
+    @abstractmethod
+    def _enrich_response_handler(self):
```

Comment:
In the base _validate_and_post_to_api method, these are not handler, it's validate. So change to _enrich_response_validator.